src/components/SearchField.tsx

```diff
@@ -1,38 +1,241 @@
 
-  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+  const [focus, setFocus] = useState<boolean>(false);
+
+  const [dropdown, setDropdown] = useState<boolean>(false);
+
+  const [dateRange, setDateRange] = useState<{
+    fromDate: number;
+    toDate: number;
+  }>({
+    fromDate: 1980,
+    toDate: 2021
+  });
+
+  const [checkboxes, setCheckboxes] = useState<{
+    favorites: boolean;
+    history: boolean;
+  }>({
+    favorites: false,
+    history: false
+  });
+
+  const [genreToSearch, setGenreToSearch] = useState<string>('none');
+
+  const searchInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setInputValue(event.target.value);
   };
 
-  const submitHandler = (event: React.SyntheticEvent) => {
-    event.preventDefault();
-    if (inputValue.trim()) {
-      console.log(inputValue);
-    }
-  };
+  const submitSearchRequest = useCallback(
+    (event: React.SyntheticEvent) => {
+      event.preventDefault();
+
+      // for proper using of query params need to refactor routing
+      if (inputValue.trim()) {
+        history.push(
+          `/search-results/title=${inputValue}?genre=${genreToSearch}?fromDate=${dateRange.fromDate}?toDate=${dateRange.toDate}?favorites=${checkboxes.favorites}?history=${checkboxes.history}`
+        );
+      }
+    },
+    [inputValue, history, checkboxes, genreToSearch, dateRange]
+  );
+
+  const onFocusSearchInput = useCallback(
+    (event: React.SyntheticEvent): void => {
+      setFocus(true);
+    },
+    []
+  );
+
+  const onBlurSearchInput = useCallback((event: React.SyntheticEvent): void => {
+    setFocus(false);
+  }, []);
+
+  const mouseEnterDropdown = useCallback(
+    (event: React.SyntheticEvent): void => {
+      setDropdown(true);
+    },
+    []
+  );
+
+  const mouseLeaveDropdown = useCallback(
+    (event: React.SyntheticEvent): void => {
+      setDropdown(false);
+    },
+    []
+  );
+
+  const fromDateInputIncrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.fromDate >= dateRange.toDate) return;
+      setDateRange({ ...dateRange, fromDate: dateRange.fromDate + 1 });
+    },
+    [dateRange]
+  );
+
+  const fromDateInputDecrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.fromDate <= 1980) return;
+      setDateRange({ ...dateRange, fromDate: dateRange.fromDate - 1 });
+    },
+    [dateRange]
+  );
+
+  const toDateInputIncrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.toDate >= 2022) return;
+      setDateRange({ ...dateRange, toDate: dateRange.toDate + 1 });
+    },
+    [dateRange]
+  );
+
+  const toDateInputDecrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.toDate <= dateRange.fromDate) return;
+      setDateRange({ ...dateRange, toDate: dateRange.toDate - 1 });
```

Comment:
toDate-1.... See comment about renaming to toYear

Reply:
done